cpu/samd5x/periph/can.c

```diff
@@ -0,0 +1,594 @@
+
+    int16_t idx = 0;
+    bool _filter_exists = false;
+    if (filter->can_id & CAN_EFF_FLAG) {
+        _filter_exists = _find_filter(dev, filter, false, &idx);
+        if (_filter_exists) {
+            DEBUG("Extended filter to disable at idx = %d\n", idx);
+            dev->msg_ram_conf.ext_filter[idx].XIDFE_0.bit.EFEC = CAN_FILTER_DISABLE;
+
+            for (uint8_t i = 0; i < ARRAY_SIZE(dev->msg_ram_conf.ext_filter); i++) {
+                DEBUG("can->msg_ram_conf.std_filter[%u] = 0x%08lx, filter conf = %u\n", i,
+                        (uint32_t)(dev->msg_ram_conf.ext_filter[i].XIDFE_0.bit.EFID1), dev->msg_ram_conf.ext_filter[i].XIDFE_0.bit.EFEC);
+            }
+        }
+        else {
+            DEBUG_PUTS("Filter not found");
+            return -1;
+        }
+    }
+    else {
+        _filter_exists = _find_filter(dev, filter, true, &idx);
+        if(_filter_exists) {
+            DEBUG("Standard filter to disable at idx = %d\n", idx);
+            dev->msg_ram_conf.std_filter[idx].SIDFE_0.bit.SFEC = CAN_FILTER_DISABLE;
+        }
+        else {
+            DEBUG_PUTS("Filter not found");
+            return -1;
+        }
+    }
+
+    return idx;
+}
+
+static void _isr(candev_t *candev)
+{
+    can_t *dev = container_of(candev, can_t, candev);
+
+    uint32_t irq_reg = dev->conf->can->IR.reg;
+    if (irq_reg & CAN_IR_TSW) {
+        DEBUG_PUTS("Timestamp wraparound interrupt");
+        /* Clear the interrupt source flag */
+        dev->conf->can->IR.reg |= CAN_IR_TSW;
+    }
+
+    if (irq_reg & CAN_IR_RF0N) {
+        DEBUG_PUTS("New message in Rx FIFO 0");
+        /* Clear the interrupt source flag */
+        dev->conf->can->IR.reg |= CAN_IR_RF0N;
+
+        uint16_t rx_get_idx = 0;
+        uint16_t rx_put_idx = 0;
+        rx_get_idx = (dev->conf->can->RXF0S.reg >> CAN_RXF0S_F0GI_Pos) & 0x3F;
+        DEBUG("rx get index = %u\n", rx_get_idx);
+        rx_put_idx = (dev->conf->can->RXF0S.reg >> CAN_RXF0S_F0PI_Pos) & 0x3F;
+        DEBUG("rx put index = %u\n", rx_put_idx);
+
+        struct can_frame frame_received = {0};
+        if (!dev->msg_ram_conf.rx_fifo_0[rx_get_idx].RXF0E_0.bit.XTD) {
+            DEBUG_PUTS("Received standard CAN frame");
+            frame_received.can_id = dev->msg_ram_conf.rx_fifo_0[rx_get_idx].RXF0E_0.bit.ID >> 18;
+        }
+        else {
+            DEBUG_PUTS("Received extended CAN frame");
+            frame_received.can_id = dev->msg_ram_conf.rx_fifo_0[rx_get_idx].RXF0E_0.bit.ID;
+        }
+        frame_received.can_dlc = dev->msg_ram_conf.rx_fifo_0[rx_get_idx].RXF0E_1.bit.DLC;
+        memcpy(frame_received.data, (uint32_t *)dev->msg_ram_conf.rx_fifo_0[rx_get_idx].RXF0E_DATA, frame_received.can_dlc);
+
+        dev->conf->can->RXF0A.reg = CAN_RXF0A_F0AI(rx_get_idx);
+        if (dev->candev.event_callback) {
+            dev->candev.event_callback(&(dev->candev), CANDEV_EVENT_RX_INDICATION, &frame_received);
+        }
+    }
+
+    if (irq_reg & CAN_IR_RF1N) {
```

Comment:
What's the difference between `CAN_IR_RF0N` and `CAN_IR_RF1N`? (comment would be nice)